### PR TITLE
Show only related tags, select multiple tags

### PR DESCRIPTION
**Pull request by bimlas**
When a tag is selected, the tag list narrows to show only the related
ones: all tags associated to the currently visible notes. Clicking on
the plus sign near another tag narrows the list again to the tags of
notes associated with the firstly AND secondly selected tag. To show
every tags again, press the tag icon on the top-left corner of
Boostnote.

Before:
screencast

After:
screencast

NOTE: Tags are joined with `&` character (`#` not works) in
`location.pathname` thus it will make the tags with this character
unavailable. Any suggestion to pass multiple values via pathname?

Please check *browser/components/TagListItem.styl*, I'm sure that it can be simpler.

### Related issues:

* https://github.com/BoostIO/Boostnote/issues/1756

### CODE REVIEW

**Comment by bimlas** [comment]
I just changed to use ` ` character to separate tags in path (`/tags/currently selected tags`) instead of `&` (`/tags/currently&selected&tags`) to prevent interference with tags containing this char (`black&white`). See commit message. I think it's stable enough to get merged.

**Comment by bimlas** [comment]
Changed the behaviour: does not narrowing the list.

screencast

**Comment by bimlas** [comment]
Just made the final version:

screencast

**Comment by bimlas** [comment]
If it's ok, then I'll try to write tests.

File: browser/main/index.js [hunk 1]
```diff
@@ -73,6 +73,9 @@ ReactDOM.render((
           <IndexRedirect to='/alltags' />
           <Route path=':tagname' />
         </Route>
+        <Route path='narrowToTag'>
```
Comment (sosukesuzuki) [inline browser/main/index.js]: What is this for?

Comment (bimlas) [reply browser/main/index.js]: Just forgot to remove, thanks for pointing it out!

**Comment by sosukesuzuki** [comment]
LGTM:smile:

**Comment by bimlas** [comment]
Should I write tests in a different PR or this one is not finished without those?

**Comment by bimlas** [comment]
I'm so sorry, but it's too hard for me: Boostnote is the very first React / Electron / whatever project for me and I'm just a beginner in the Javascript world, thus writing such tests is nearly impossible. All what I did is modifying Jest test.

**Comment by Rokt33r** [comment]
@bimlas I'll take care of it. 😄

**Comment by kazup01** [comment]
Hi @bimlas , thanks for your contribution! There is conflict, could you fix it?

**Comment by bimlas** [comment]
Rebased, fixed.

**Comment by kazup01** [comment]
Merged. Thank you for your contriburion @bimlas !

**Pull request merged**